feat(cdk8s): import an explicit k8s spec version

misterjoshua opened the pull request:
Allows users to specify the `k8s` version to import.

Fixes #1557 

---
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

Chriscbr commented on src/cdk8s/cdk8s-app-ts.ts:
```diff
@@ -12,6 +12,13 @@ export interface Cdk8sTypeScriptAppOptions extends TypeScriptProjectOptions {
+   *
+   * @default - Use the cdk8s default
+   */
+  readonly cdk8sApiVersion?: string;
```
This name seems misleading, since it's not the version of cdk8s that is changing. How about `k8sVersion`?

misterjoshua replied:
`k8sSpecVersion` sounds good.

Chriscbr commented on src/cdk8s/cdk8s-app-ts.ts:
```diff
@@ -173,9 +180,13 @@ export class Cdk8sTypeScriptApp extends TypeScriptAppProject {
       exec: "cdk8s synth",
     });
 
+    const k8sSpec = options.cdk8sApiVersion
+      ? `k8s@${options.cdk8sApiVersion}`
+      : "k8s";
+
     const importTask = this.addTask("import", {
       description: "Imports API objects to your app by generating constructs.",
-      exec: "cdk8s import -o src/imports",
+      exec: `cdk8s import ${k8sSpec} -o src/imports`,
```
I'm not sure if this is the ideal place to specify the k8s import. By default, `cdk8s import` (without specifying any spec files) will import all specs listed in `cdk8s.yaml` ([docs](https://cdk8s.io/docs/latest/cli/import/#crds)). Instead of updating this command, projen should auto-generate cdk8s.yaml with the correct list of specs instead.

misterjoshua replied:
I believe the app creates a `cdk8s.yaml` as `SampleCode`. I wonder if we should bring this file into projen management so that we can control the file via the project options?

Chriscbr replied:
That sounds good 👍

misterjoshua commented:
@Chriscbr Hey. I've changed the new option to `k8sSpecVersion`, and I've added a `YamlFile` to generate `cdk8s.yaml`. Ready for another review when you get a chance.

misterjoshua commented on src/cdk8s/cdk8s-app-ts.ts:
```diff
@@ -173,21 +181,31 @@ export class Cdk8sTypeScriptApp extends TypeScriptAppProject {
-      importTask.exec(`cdk8s import ${importSpec} -o src/imports`);
-    }
-
-    this.gitignore.include("imports/");
```
This line seems like it was left dangling from #1536, since now the imports are `src/imports` by default, so I removed it.

Chriscbr reviewed (approved):
Nice! 👍
